overlord/configstate/config/helpers.go

```diff
@@ -119,6 +120,96 @@ func GetFromChange(snapName string, subkeys []string, pos int, config map[string
 	return GetFromChange(snapName, subkeys, pos+1, configm, result)
 }
 
+// SaveRevisionConfigMaybe makes a copy of config -> snapSnape configuration into the versioned config.
+// It doesn't do anything if there is no configuration for given snap in the state.
+// The caller is responsible for locking the state.
+func SaveRevisionConfigMaybe(st *state.State, snapName string, rev snap.Revision) error {
```

Comment:
Those would be nicer without the "maybe", per prior note. All configuration handling deals with the empty and missing configs as a fine case. It's just an optimization to not store it if empty.